packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -205,14 +211,59 @@ export class DataStores implements IDisposable {
                 }),
             pkg);
 
-        // Resolve pending gets and store off any new ones
         this.contexts.addBoundOrRemoted(remotedFluidDataStoreContext);
 
         // Equivalent of nextTick() - Prefetch once all current ops have completed
         // eslint-disable-next-line @typescript-eslint/no-floating-promises
         Promise.resolve().then(async () => remotedFluidDataStoreContext.realize());
     }
 
+    public processAliasMessage(message: ISequencedDocumentMessage): IDataStoreAliasMapping | undefined {
+        const aliasMessage = message.contents as IDataStoreAliasMessage;
+        if (!isDataStoreAliasMessage(aliasMessage)) {
+            throw new DataCorruptionError(
+                "malformedDataStoreAliasMessage",
+                {
+                    ...extractSafePropertiesFromMessage(message),
+                },
+            );
+        }
+
+        const existingMapping = this.aliasMap.get(aliasMessage.alias);
+        if (existingMapping !== undefined) {
+            return {
+                suppliedInternalId: aliasMessage.internalId,
+                alias: aliasMessage.alias,
+                aliasedInternalId: existingMapping,
+            };
+        }
+
+        // Unlikely scenario, but we may have an alias OP with the alias value
+        // equal to the id of an older named data store
+        const maybeContextWithAliasAsId = this.contexts.get(aliasMessage.alias);
+        if (maybeContextWithAliasAsId !== undefined) {
+            return {
+                suppliedInternalId: aliasMessage.internalId,
+                alias: aliasMessage.alias,
+                aliasedInternalId: maybeContextWithAliasAsId.id,
+            };
+        }
+
+        const currentContext = this.contexts.get(aliasMessage.internalId);
+        if (currentContext === undefined) {
+            return undefined;
```

Comment:
This seems like should be impossible to get to (unless we messed up with GC). I'd rather learn about it from telemetry than silently drop this info on the floor

Reply:
will make it send an event here..